packages/block-editor/src/components/link-control/index.js

```diff
@@ -43,16 +43,18 @@ const MODE_EDIT = 'edit';
 
 function LinkControl( {
 	className,
-	value,
+	value = {},
```

Comment:
What properties can this object have? Can they not be normal props instead of nested props?

Reply:
right now, url, title, openInNewTab but it supports custom settings adding custom properties there.

Reply:
1. This should be documented (granted, not really the responsibility of the refactor)
2. In this current implementation, there's some added chance of extra work on the React reconciler, since the value is [passed through to `LinkControlSettingsDrawer`](https://github.com/WordPress/gutenberg/blob/2d1add52ec4f179b5e2fca0814a48c255f84da2b/packages/block-editor/src/components/link-control/index.js#L236), and it would be different each time if not provided explicitly (not a consistent default object reference).

Reply:
This is a cause of some confusion.

The documentation refers to this as an "item", and the example includes some reference to an `id` property:


https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/link-control/README.md#onchange

Presumably, this is referring to some post object, like in the underlying URLInput `onChange` second argumnet:

https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/url-input#onchange-url-string-post-object--function

However: Even for `URLInput`, this value is acknowledged to be _null unless the user selects a post_. So, what should the value be for when a user chooses to enter a raw URL? An `id` won't exist, because there's no post associated with this URL.

@youknowriad , based on your comment, we could choose to make some custom shape which is an amalgamation of a post and various other link-specific values (`opensInNewTab` appears to be one).

>right now: url, title, openInNewTab but it supports custom settings adding custom properties there.

On this line of thinking, we should:

- Be very specific with what these are (I think the set you mention is agreeable)
- Explicitly _omit_ any other property values. Right now, we're [spreading post properties](https://github.com/WordPress/gutenberg/blob/b55605e32180b874ef34afa86f830fe0fa1bef7f/packages/block-editor/src/components/link-control/index.js#L166) into the changed value. I don't think we should want to set any expectation that these values will be available, as otherwise it becomes very error-prone when the user enters a custom URL which is not associated with a post.
- Fix any inaccuracies (i.e. remove the reference to `item.id` in the current documentation)

Reply:
@aduth This makes a lot of sense to me, I wasn't aware that we had these extra properties added there as well.